### PR TITLE
Add support for Idempotency-Key header

### PR DESCRIPTION
**Pull request by VaclavK**
added code and tests for new idempotency key header derived off  SB message Id
not sent for GET requests

### CODE REVIEW

File: src/CaptainHook.Fabric/ApplicationParameters/Local.1Node.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Application xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="fabric:/CaptainHook" xmlns="http://schemas.microsoft.com/2011/01/fabric">
   <Parameters>
-    <Parameter Name="KeyVault_Base_Uri" Value="https://esw-tooling-ci-we.vault.azure.net/" />
+    <Parameter Name="KeyVault_Base_Uri" Value="https://esw-tooling-ci-we-oh2.vault.azure.net/" />
```
Comment (mszymeshop) [inline src/CaptainHook.Fabric/ApplicationParameters/Local.1Node.xml]: This should not get checked in, right?

Comment (VaclavK) [reply src/CaptainHook.Fabric/ApplicationParameters/Local.1Node.xml]: I unchecked the thing but github desktop decided to be smarter... cheers

File: src/CaptainHook.EventHandlerActor/Handlers/RequestBuilder.cs [hunk 2]
```diff
@@ -251,6 +251,11 @@ public WebHookHeaders GetHttpHeaders(WebhookConfig webhookConfig, MessageData me
             headers.AddRequestHeader(Constants.Headers.EventDeliveryId, messageData.CorrelationId);
             headers.AddRequestHeader(Constants.Headers.EventType, webhookConfig.PayloadTransformation==PayloadContractTypeEnum.WrapperContract ? typeof(NewtonsoftDeliveryStatusMessage).FullName.ToLowerInvariant(): webhookConfig.EventType);
 
+            if (webhookConfig.HttpMethod!=HttpMethod.Get)
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/CaptainHook.EventHandlerActor/Handlers/RequestBuilder.cs]: the ticket says not to - see exception criteria and it made sense

Not present for safe verbs. i.e. GET 

I even considered DELETE but there it makes sense (e.g. soft delete flows)

I am happy to amend the ticket....

[the remark this answers is not in the document]
Comment (VaclavK) [reply src/CaptainHook.EventHandlerActor/Handlers/RequestBuilder.cs]: @dguerin have a  look at the latest

**Comment by dguerin** [comment]
there is one other item I need to investigate but I should have an answer to this tomorrow before lunch.

**Pull request merged**